### ISSUE TITLE
Nether and End inaccesible after updating

### ISSUE THREAD

**Issue by EnzoGaGu** [LunaPixelStudios/Better-MC#1575]
### Describe the Bug

After updating to the last version of BMC4, the Nether and End are inaccesible in previously created worlds. Both portals generate but don't work (endless swirling in the nether one, and the End portal just doesn't teleport you anywhere). You can't even use the /execute in command, because it doesn't find those dimensions. This doesn't happen with worlds created in v26.

### Reproduction Steps

1. Update the modpack to the last version
2. Try to enter the Nether or End
3. Portals don't work
...

### Screenshots and Videos

![imagen_2024-02-17_135602632](https://github.com/LunaPixelStudios/Better-MC/assets/96706572/e624db0c-f4d6-4b35-a496-0534ea88eb06)
![imagen_2024-02-17_135620455](https://github.com/LunaPixelStudios/Better-MC/assets/96706572/7d260256-e9d1-49f9-846a-05e5d22ef67c)
![imagen_2024-02-17_135855974](https://github.com/LunaPixelStudios/Better-MC/assets/96706572/3ffa751f-6f58-49fc-9b7a-43c96aa6f9f8)

### Operating System

Windows 10 

### Minecraft Version

1.20

### Modloader

Forge

### Modpack Version

BMC v26

### Other Mods

Just the BSL shaders.

**Comment by chorbintime** [comment]
Are you on a server? If so, how is it hosted?

**Comment by hacktic-dev** [comment]
I can reproduce this. Server hosted via Shockbyte

**Comment by EnzoGaGu** [comment]
No, I'm not! I tried opening the world to LAN to see if that somehow fixed the problem, but it continued doing the exact same thing

**Comment by chorbintime** [comment]
Try regenerating the dimension files for the Nether and End?

**Comment by EnzoGaGu** [comment]
I tried, but nothing changed. I deleted the DIM1 and DIM-1 folders and they never generated again (portals continued doing the exact same thing as before). I tried deleting the Region folder inside them instead of the whole thing but it didn't worked either way. 

It seems like the specific world doesn't associate the portals to the specific dimensions/don't know those dimensions exist, if that makes sense?

**Comment by chorbintime** [comment]
Can you send over your world file? Use google drive or mediafire

**Comment by EnzoGaGu** [comment]
[https://drive.google.com/file/d/1j7vjK_OtwMAHkiWORzC4AS9-1EJslJoO/view?usp=sharing](url)


There's a .rar with 2 worlds. New World is the one from where the pictures on this post came from, and the other is my main survival world where I first noticed this bug.

**Comment by chorbintime** [comment]
Did a lot of testing on this and honestly can not pinpoint a cause - it could be a JER and Repurposed Structures conflict of some kind but loading into the world after removing those just gives a datapack validation error that makes the world unjoinable. You may need to stay on v25, or try v27 when it releases. I apologize for the inconveinence.

You could also try adding only the mods from v26 and see if the issue still happens on your v25 world, or try updating only some of the mods. Let me know the results if you do this. Thank you

**Comment by EnzoGaGu** [comment]
No problems! Thanks a lot for the effort. These kind of stuff happen.

I'll keep you updated if I come across any solutions!

**Comment by EnzoGaGu** [comment]
So, after all this time, I finally found a solution for this problem. If someone is facing the same issue, at least in my case it seems like level.dat, level.dat_old and/or session.lock got corrupted at some point. 

I just grabbed those files from an early backup of the same world, replaced the corrupted ones with those and it started working fine. I lost my inventory but that's nothing compared to losing the whole world lol
